aesara/link/c/c_code/lazylinker_c.c

```diff
@@ -18,52 +18,79 @@
 
 TODO:
 - Check max supported depth of recursion
-- CLazyLinker should add context information to errors caught during evaluation. Say what node we were on, add the traceback attached to the node.
+- CLazyLinker should add context information to errors caught during evaluation.
+Say what node we were on, add the traceback attached to the node.
 - Clear containers of fully-useed intermediate results if allow_gc is 1
 - Add timers for profiling
 - Add support for profiling space used.
 
 
   */
-static double pytime(const struct timeval * tv)
-{
+static double pytime(const struct timeval *tv) {
   struct timeval t;
-  if (!tv)
-    {
-      tv = &t;
-      gettimeofday(&t, NULL);
-    }
-  return (double) tv->tv_sec + (double) tv->tv_usec / 1000000.0;
+  if (!tv) {
+    tv = &t;
+    gettimeofday(&t, NULL);
+  }
+  return (double)tv->tv_sec + (double)tv->tv_usec / 1000000.0;
 }
 
 /**
   Helper routine to convert a PyList of integers to a c array of integers.
   */
-static int unpack_list_of_ssize_t(PyObject * pylist, Py_ssize_t **dst, Py_ssize_t *len,
-                                  const char* kwname)
-{
+static int unpack_list_of_ssize_t(PyObject *pylist, Py_ssize_t **dst,
+                                  Py_ssize_t *len, const char *kwname) {
   Py_ssize_t buflen, *buf;
-  if (!PyList_Check(pylist))
-    {
-      PyErr_Format(PyExc_TypeError, "%s must be list", kwname);
+  if (!PyList_Check(pylist)) {
+    PyErr_Format(PyExc_TypeError, "%s must be list", kwname);
+    return -1;
+  }
+  assert(NULL == *dst);
+  *len = buflen = PyList_Size(pylist);
+  *dst = buf = (Py_ssize_t *)calloc(buflen, sizeof(Py_ssize_t));
```

Comment:
How does this get freed?

Reply:
It should be freed when the Python VM calls `CLazyLinker_dealloc`.